rioxarray/_io.py

```diff
@@ -194,14 +195,49 @@ def parsevec(s):
     return parsed_tags
 
 
+def build_subdataset_filter(group_names, variable_names):
+    """
+    Examples:
+'HDF4_EOS:EOS_GRID:"./modis/MOD09GQ.A2017290.h11v04.006.NRT.hdf":MODIS_Grid_2D:sur_refl_b01_1'        
+ 
+    Parameters
+    ----------
+    group_names: str or list or tuple
+        Name or names of netCDF groups to filter by.
+
+    variable_names: str or list or tuple
+        Name or names of netCDF variables to filter by.
+ 
+    Returns
+    -------
+    re.SRE_Pattern: output of re.compile()
+    """
+    variable_query = r"\w+"
+    if variable_names is not None:
+        if not isinstance(variable_names, (tuple, list)):
+            variable_names = [variable_names]
+        variable_query = rf"(?:{'|'.join(variable_names)})"
```

Comment:
Are there any restrictions on the syntax of `variable_names`? Might want to run them through [`re.escape`](https://docs.python.org/3/library/re.html#re.escape).

Reply:
Good suggestion. Added it in. Thanks!